### PR TITLE
Onelogin merge

### PR DESCRIPTION
**Pull request by pvorozheykina**

### CODE REVIEW

File: lib/onelogin/ruby-saml/response.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@
 require "time"
 require "nokogiri"
 
+require "pry"
```
Comment (curious-attempt-bunny) [inline lib/onelogin/ruby-saml/response.rb]: We don't want to ship this line.

File: lib/onelogin/ruby-saml/response.rb [hunk 2]
```diff
@@ -560,40 +555,28 @@ def validate_audience
...
 
-        if not_before && (now + allowed_clock_drift) < not_before
+        if not_before && (now + allowed_clock_drift) < not_before 
           error_msg = "Current time is earlier than NotBefore condition #{(now + allowed_clock_drift)} < #{not_before})"
-          return append_error(error_msg)
+          return append_error(error_msg, soft)
         end
-
-        if not_on_or_after && now >= (not_on_or_after + allowed_clock_drift)
+        #binding.pry
```
Comment (curious-attempt-bunny) [inline lib/onelogin/ruby-saml/response.rb]: Ditto.

File: lib/onelogin/ruby-saml/response.rb [hunk 3]
```diff
@@ -560,40 +555,28 @@ def validate_audience
...
         now = Time.now.utc
 
-        if not_before && (now + allowed_clock_drift) < not_before
+        if not_before && (now + allowed_clock_drift) < not_before 
           error_msg = "Current time is earlier than NotBefore condition #{(now + allowed_clock_drift)} < #{not_before})"
-          return append_error(error_msg)
+          return append_error(error_msg, soft)
         end
-
-        if not_on_or_after && now >= (not_on_or_after + allowed_clock_drift)
```
Comment (curious-attempt-bunny) [inline lib/onelogin/ruby-saml/response.rb]: Didn't we want to keep this `+ allowed_clock_drift`?

Comment (curious-attempt-bunny) [inline lib/onelogin/ruby-saml/response.rb]: Polina and I discussed this and decided to leave it as is.

**Comment by pvorozheykina** [comment]
@curious-attempt-bunny removed the debuggers and commented out code.

**Comment by curious-attempt-bunny** [comment]
Looking good.

**Pull request merged**